### PR TITLE
Utils: Local minima/maxima for 1-d array values

### PR DESCRIPTION
**Pull request by ntejos**
Added `is_local_minima_maxima()` and `is_local_minima_maxima()` utilities for 1-d array analysis. Test added too.

### CODE REVIEW

File: linetools/utils.py [hunk 1]
```diff
@@ -581,4 +581,60 @@ def overlapping_chunks(chunk1, chunk2):
...
         return False
     else:  # overlap exists
-        return True
\ No newline at end of file
+        return True
+
+
+def is_local_minima(a):
+    """For a given 1-d array a, it returns True for local minima
+    and False otherwise. Both edges of array are always False by
```
Comment (profxj) [inline linetools/utils.py]: Describe in the doc what defines a 'local minima'.

File: linetools/utils.py [hunk 2]
```diff
@@ -581,4 +581,60 @@ def overlapping_chunks(chunk1, chunk2):
...
+    for i in range(1, len(a) - 1):  # note that the two edges are always False by definition
+        cond = (a[i] < a[i - 1]) and (a[i] < a[i + 1])
+        if cond: # local minima
+            mask[i] = True
+    return mask
+
+
+def is_local_maxima(a):
+    """For a given 1-d array a, it returns True for local maxima
+    and False otherwise. Both edges of array are always False by
```
Comment (profxj) [inline linetools/utils.py]: Same as above for local minimum

File: linetools/utils.py [hunk 3]
```diff
@@ -581,4 +581,60 @@ def overlapping_chunks(chunk1, chunk2):
...
+    Returns
+    -------
+    boolean, whether the value is a local minima
+
+    Notes
+    -----
+    See also is_local_maxima()
+
+    """
+    a = np.array(a)
```
Comment (profxj) [inline linetools/utils.py]: Is this necessary?

File: linetools/utils.py [hunk 4]
```diff
@@ -581,4 +581,60 @@ def overlapping_chunks(chunk1, chunk2):
...
+    mask = np.zeros_like(a).astype(bool)
+    for i in range(1, len(a) - 1):  # note that the two edges are always False by definition
```
Comment (profxj) [inline linetools/utils.py]: Can we avoid a FOR loop?

I bet we can, e.g.

b = a - np.roll(a,1)
c = a - np.roll(a,-1)

Now do some logic on b and c...

File: linetools/utils.py [hunk 5]
```diff
@@ -581,4 +581,60 @@ def overlapping_chunks(chunk1, chunk2):
...
+    boolean, whether the value is a local maxima
+
+    Notes
+    -----
+    See also is_local_minima()
+
+    """
+    a = np.array(a)
+    mask = np.zeros_like(a).astype(bool)
+    for i in range(1, len(a) - 1):  # note that the two edges are always False by definition
```
Comment (profxj) [inline linetools/utils.py]: Same issue as above (avoid FOR loop)

File: linetools/utils.py (hunk 1 above)
Comment (ntejos) [inline linetools/utils.py]: done.

File: linetools/utils.py (hunk 3 above)
Comment (ntejos) [inline linetools/utils.py]: Apparently not.

File: linetools/utils.py (hunk 4 above)
Comment (ntejos) [inline linetools/utils.py]: Good idea.

File: linetools/utils.py (hunk 2 above)
Comment (ntejos) [inline linetools/utils.py]: Sure.

File: linetools/utils.py (hunk 5 above)
Comment (ntejos) [inline linetools/utils.py]: Sure.

**Comment by profxj** [comment]
Merging.

**Pull request merged**